profile/filter_test.go

```diff
@@ -0,0 +1,252 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package profile
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/google/pprof/internal/proftest"
+)
+
+func TestFilter(t *testing.T) {
```

Comment:
I think you could (and should for clarity) combine TestFilter and TestFilter2 by checking fm, im, hm, and hmn in TestFilter2.

Reply:
I've added checks for the return value, and removed the original test function since its now redundant.